distributed/nanny.py

```diff
@@ -87,6 +87,19 @@ class Nanny(ServerNode):
             2. Existing environment variables
             3. Dask configuration
 
+        Note
+        ----
+        Some environment variables, like ``OMP_NUM_THREADS``, must be set before
+        importing numpy to have effect. Others, like ``MALLOC_TRIM_THRESHOLD_`` (see
+        :ref:`memtrim`), must be set before starting the Linux process. So we need to
+        set them before spawning the subprocess, even if this means poisoning the
+        process running the Nanny.
+
+        For the same reason, be warned that changing
+        ``distributed.worker.multiprocessing-method`` from ``spawn`` to ``fork`` or
```

Comment:
I get the problems with forkserver, but why fork? The variables should be set before the fork happens, right?

Reply:
It's because the `MALLOC_TRIM_THRESHOLD_` variable is read when the glibc initialises itself, and in case of fork it's already initialised